coda-src/kerndep/coda.h

```diff
@@ -149,13 +149,29 @@ struct venus_dirent {
 
 #ifndef _VENUS_DIRENT_T_
 #define _VENUS_DIRENT_T_ 1
+#if defined(__FreeBSD__)
+#include <sys/param.h>
+#endif
+#if defined(__FreeBSD__) && (__FreeBSD_version >= 1200031)
```

Comment:
It's a hack indeed - but it used to work, and this change just brings it back to the state it was before.  Practically speaking, it should be fine - the change to "struct dirent" change in FreeBSD 12 is expected to break some of the things.

It's true that the proper way would be to do the translation in the kernel module, but that's quite a bit of work.  The way it works now it just reads (calls VOP_READ) the backing file and returns that as the directory contents.